basculehttp/log.go

```diff
@@ -69,11 +73,48 @@ func sanitizeHeaders(headers http.Header) (filtered http.Header) {
 func SetLogger(logger *zap.Logger) alice.Constructor {
 	return func(delegate http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			r = r.WithContext(sallust.With(r.Context(),
-				logger.With(
-					zap.Reflect("requestHeaders", sanitizeHeaders(r.Header)), //lgtm [go/clear-text-logging]
-					zap.String("requestURL", r.URL.EscapedPath()),
-					zap.String("method", r.Method))))
+			var tid string
+
+			if tid = r.Header.Get(transaction.HeaderWPATID); tid == "" {
+				tid = transaction.GenTID()
+			}
+
+			var satClientID = "N/A"
+
+			// retrieve satClientID from request context
+			if auth, ok := bascule.FromContext(r.Context()); ok {
```

Comment:
SetLogger sets the logger before bascule adds its Auth to the context, so this won't give us anything.  In order to add this to the logger, we need to do it later:
1. add a logger with most things to the context (this function).
2. run bascule middleware.
3. pull the logger out and add anything from the bascule context information.
4. application logic.

To accomplish 3, we can make a new function similar to this one in this file that:
1. pulls the logger out of the context
2. pulls the bascule Authentication out of the context
3. adds any bascule information to the logger
4. puts the logger back in the context

Its function signature should return an `alice.Constructor`, similar to `SetLogger()`.